docs/AnnotationSpec.md

```diff
@@ -1,58 +1,108 @@
 
-    '''@nni.report_final_result(test_acc)'''
+Note that every annotation should be followed by an assignment statement.
 
- 5. Annotation `function_choice` in code as:
+In this way, users could either run the python code directly or launch nni to tune hyper-parameter in this code, without changing any codes.
 
-    '''@nni.function_choice(max_pool(h_conv1, self.pool_size),avg_pool(h_conv1, self.pool_size),name=max_pool)'''
 
-In this way, they can easily implement automatic tuning on NNI. 
+## Types
 
-For `@nni.variable`, `nni.choice` is the type of search space and there are 10 types to express your search space as follows:
+### Types of Annotation:
 
- 1. `@nni.variable(nni.choice(option1,option2,...,optionN),name=variable)`  
-    Which means the variable value is one of the options, which should be a list The elements of options can themselves be stochastic expressions
+In NNI, there are mainly four types of annotation:
 
- 2. `@nni.variable(nni.randint(upper),name=variable)`  
-    Which means the variable value is a random integer in the range [0, upper).
 
- 3. `@nni.variable(nni.uniform(low, high),name=variable)`  
-    Which means the variable value is a value uniformly between low and high.
+ 1. Annotate variables in code as:
 
- 4. `@nni.variable(nni.quniform(low, high, q),name=variable)`  
-    Which means the variable value is a value like round(uniform(low, high) / q) * q
+    `'''@nni.variable(nni.choice(2,3,5,7),name=self.conv_size)'''`
 
- 5. `@nni.variable(nni.loguniform(low, high),name=variable)`  
-    Which means the variable value is a value drawn according to exp(uniform(low, high)) so that the logarithm of the return value is uniformly distributed.
+ 2. Annotate functions in code as:
 
- 6. `@nni.variable(nni.qloguniform(low, high, q),name=variable)`  
-    Which means the variable value is a value like round(exp(uniform(low, high)) / q) * q
+    `'''@nni.function_choice(max_pool(h_conv1, self.pool_size), avg_pool(h_conv1, self.pool_size), name=max_pool)'''`
 
- 7. `@nni.variable(nni.normal(label, mu, sigma),name=variable)`  
-    Which means the variable value is a real value that's normally-distributed with mean mu and standard deviation sigma.
+ 3. Annotate intermediate result in code as:
 
- 8. `@nni.variable(nni.qnormal(label, mu, sigma, q),name=variable)`  
-    Which means the variable value is a value like round(normal(mu, sigma) / q) * q
+    `'''@nni.report_intermediate_result(test_acc)'''`
 
- 9. `@nni.variable(nni.lognormal(label, mu, sigma),name=variable)`  
-    Which means the variable value is a value drawn according to exp(normal(mu, sigma))
+ 4. Annotate final result in code as:
+
+    `'''@nni.report_final_result(test_acc)'''`
+
+
+### Types of Search Space
+
+For `@nni.variable`, **`nni.choice`** is the type of search space and there are 10 types to express your search space as follows:
```

Comment:
Shall we out the types of Search Space to another file? Because it not only design for annotation.